rholang/src/main/scala/coop/rchain/rholang/interpreter/Substitute.scala

```diff
@@ -23,6 +24,13 @@ object Substitute {
       bSub <- evB.substitute(termB)
     } yield f(aSub, bSub)
 
+  def substituteNoSort2[M[_]: Monad, A, B, C](termA: A, termB: B)(
+      f: (A, B) => C)(implicit evA: Substitute[M, A], evB: Substitute[M, B], env: Env[Par]): M[C] =
+    for {
+      aSub <- evA.substituteNoSort(termA)
+      bSub <- evB.substituteNoSort(termB)
+    } yield f(aSub, bSub)
```

Comment:
When implementing `substitute2` I forgot `Applicative should be enough for this. Could you refactor both of the function to:
```
def substitute2[M[_]: Applicative, A, B, C](termA: A, termB: B)(
      f: (A, B) => C)(implicit evA: Substitute[M, A], evB: Substitute[M, B], env: Env[Par]): M[C] =
    (evA.substitute(termA), evB.substitute(termB)).mapN(f)
```
?